client/core/core.go

```diff
@@ -326,6 +326,11 @@ func (dc *dexConnection) parseMatches(msgMatches []*msgjson.Match, checkSigs boo
 		} else {
 			match.msgMatches = append(match.msgMatches, msgMatch)
 		}
+
+		status := order.MatchStatus(msgMatch.Status)
+		log.Debugf("registering match %v for order %v in status %v", msgMatch.MatchID, oid, status)
```

Comment:
Note that the server-side connect fixes in https://github.com/decred/dcrdex/pull/531 are necessary for both match messages to be sent on connect.